Author: Robert James Clay <jame@rocasa.us>
Description: Error at flock.c:43 - call to open missing mode declared with attribute
error:  open with O_CREAT in second argument needs 3 arguments.

In Ubuntu, this issue is the result of an enforcment of using the correct form
of the open() call:  reference https://wiki.ubuntu.com/CompilerFlags.
--- mbsebbs-0.92.0.orig/mbfido/flock.c
+++ mbsebbs-0.92.0/mbfido/flock.c
@@ -40,7 +40,7 @@
 	struct stat st;
 
 	if (fn) {
-		if ((lfd = open(fn,O_RDWR | O_CREAT)) < 0) {
+		if ((lfd = open(fn,O_RDWR | O_CREAT,S_IRUSR|S_IWUSR|S_IRGRP|S_IWGRP)) < 0) {
 			perror("");
 			WriteError("Error opening file %s", fn);
 			return -1;
